src/profiler.h

```diff
@@ -69,23 +69,11 @@ class Profiler
 	void avg(const std::string &name, float value)
 	{
 		JMutexAutoLock lock(m_mutex);
-		{
-			std::map<std::string, int>::iterator n = m_avgcounts.find(name);
-			if(n == m_avgcounts.end())
-				m_avgcounts[name] = 1;
```

Comment:
What is "this"?

I assume you mean initialize to 1?

int& count = m_avgcounts[name]; // Will initialise to 0
assert(count != -2);
count = MYMAX(count, 0) + 1; // Will get incremented to 1